pkg/operator/ceph/cluster/mon/drain.go

```diff
@@ -52,22 +54,45 @@ func (c *Cluster) reconcileMonPDB() error {
 }
 
 func (c *Cluster) createOrUpdateMonPDB(maxUnavailable int32) (controllerutil.OperationResult, error) {
-	pdb := &policyv1beta1.PodDisruptionBudget{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      monPDBName,
-			Namespace: c.Namespace,
-		}}
+	k8sVersion, err := k8sutil.GetK8SVersion(c.context.Clientset)
+	if err != nil {
+		return controllerutil.OperationResultNone, errors.Wrap(err, "failed to fetch get k8s version")
+	}
+	// minimum k8s version required for v1 PodDisruptionBudget is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0', else apply v1beta1 PodDisruptionBudgetJob.
+	minVersionForPDBV1 := "1.21.0"
+	usePDBV1 := k8sVersion.AtLeast(version.MustParseSemantic(minVersionForPDBV1))
```

Comment:
Could you call the method `usePDBV1Version()` here?

Reply:
but it needs [ReconcileClusterDisruption](https://pkg.go.dev/github.com/rook/rook/pkg/operator/ceph/disruption/clusterdisruption#ReconcileClusterDisruption) receiver ,so not callable, has type [cluster](https://pkg.go.dev/github.com/rook/rook/pkg/operator/ceph/cluster/mon#Cluster) receiver in the method.

Reply:
How about changing the `usePDBV1Version()` to take the clientset as the param instead? Then it can be called from more places like this.

Reply:
Done